src/segmented_buffer.rs

```diff
@@ -100,7 +100,7 @@ impl Buf for Buffer {
 
 // TODO: expose size when const generics become available
 #[derive(PartialEq)]
-pub struct SegmentedBuf<T> {
+pub struct SegmentedBuf<T: 'static> {
```

Comment:
Hmm, I had a look at the countme code and it looks like we can avoid this by explicitly setting

```rust
_c: countme::Count<SegmentedBuf<T>>
```

rather than using Self

Reply:
Unfortunately, I get the same error with `_c: countme::Count<SegmentedBuf<T>>`:
![image](https://user-images.githubusercontent.com/4270240/170767245-fde8ccce-a409-4a43-a3d6-c6122c8fc169.png)
(with either stable `Rust 1.61.0` or `v1.63.0-nightly`).